daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV2.scala

```diff
@@ -707,13 +707,11 @@ private[archive] class DecodeV2(minor: LV.Minor) {
         decodeExpr(lfChoice.getControllers, s"$tpl:$chName:controller") { controllers =>
           bindWork(
             if (lfChoice.hasObservers) {
-              assertSince(LV.Features.choiceObservers, "TemplateChoice.observers")
               decodeExpr(lfChoice.getObservers, s"$tpl:$chName:observers") { observers =>
                 Ret(Some(observers))
               }
             } else {
-              assertUntil(LV.Features.choiceObservers, "missing TemplateChoice.observers")
-              Ret(None)
+              throw notSupportedError("missing TemplateChoice.observers")
             }
```

Comment:
```suggestion        
              decodeExpr(lfChoice.getObservers, s"$tpl:$chName:observers") { observers =>
                Ret(Some(observers))
              }
```
We should be consistent with the rest. We do not check the required field are defined (e.g. line 707 we have a `lfChoice.getControllers`, without check that `lfChoice.hasControllers`) so we we should do the same here. 

I am not against we systematically verify that *all*  the required fields are actually defined, but this should be done in another PR.

Reply:
Good point, applied.